### ISSUE TITLE
Not able to find 'ALS' function

### ISSUE THREAD

**Issue by blutab** [dselivanov/reco#14]
Hi,

I am not able to find a function 'ALS' which you had used in this post:
http://dsnotes.com/post/2017-06-28-matrix-factorization-for-recommender-systems-part-2/

Kindly help.

Regards.

**Comment by dselivanov** [comment]
See `wrmf` function.

**Comment by mayankgoel4** [comment]
Hi Dmitriy,
I am working on a problem that have 0.1M Users & 0.5M products and hence was replicating your Lastfm datasets code steps for Reco package. However, while splitting data into train & validation noticed that user base is different in validation set than on which model is getting trained. So was confused how MF model will be predicting on this validation set when latent factors from trained users won't be part of validation users. Am i missing something or code is not updated?..

**Comment by dselivanov** [comment]
This is correct and very fair cross validation split. See section "Recommendations for new users" which explains who to predict items for users without embeddings http://dsnotes.com/post/2017-06-28-matrix-factorization-for-recommender-systems-part-2/

**Comment by mayankgoel4** [comment]
Thanks Dmitriy. This is awesome and package seems to be really fast for getting latent factors and worked for large volume of data.
Is it possible that you can also help me with below:
1, Any package in your portfolio that can serve as recommender based on item/user similarity based on Cosine/Jaccard etc, since standard recommenderlab package is not supporting and predicting on huge data? 
2.Any implementation that item similarities(cosine/jaccard) from latent factors can be used for further recommendations by assigining weights similar to recommenderlab package thing

**Comment by dselivanov** [comment]
@mayankgoel4 what about http://text2vec.org/similarity.html#jaccard_similarity, http://text2vec.org/similarity.html#cosine_similarity ? 
I'm not a user of recommenderlab package, so can't help. Probably the best thing to do is to ask stackoverflow or similar resource.

**Comment by mayankgoel4** [comment]
Hi Dmitriy. Just a small question. If a am using predict function of reco package, does it handle and provide recommendations for all unseen items from training datasets or i have to manually get rid of seen items outside. 
Ex: User have watched 5 items out of 100 items and we provide 10 recommendations for this user. Can any item out of 5 items that user have already watched be part of 10 recommendations or it will be provide recommendations from remaining 95(unseen) items?

**Comment by dselivanov** [comment]
See `not_recommend` argument in predict function

**Comment by mayankgoel4** [comment]
Thanks Dmitriy for your quick response and continuous support.!! This package had been of great help to us.
